### PR TITLE
Feature/ge 814

### PR DESCRIPTION
**Pull request by AltusBarry**
Pulled in django-layers, refactored template dir layout.

DONE Refactor existing middleware.
DONE Update tests

DONE get write access to django-layers
DONE Make release for django-layers

### CODE REVIEW

File: authentication_service/templates/ninyampinga/authentication_service/registration.html [hunk 1]
```diff
@@ -1,8 +1,8 @@
-{% extends "authentication_service/demo/oops.html" %}
+{% extends "authentication_service/registration.html" %}
```
Comment (AltusBarry) [inline authentication_service/templates/ninyampinga/authentication_service/registration.html]: Incorrect template name and extends. this should not be registration.

File: authentication_service/templates/ninyampinga/authentication_service/redirect_issue.html [hunk 2]
```diff
@@ -1,4 +1,4 @@
-{% extends "authentication_service/registration/registration.html" %}
+{% extends "authentication_service/redirect_issue.html" %}
```
Comment (AltusBarry) [inline authentication_service/templates/ninyampinga/authentication_service/redirect_issue.html]: Incorrect template name and extends. this should not be redirect_issue.

File: authentication_service/templates/springster/authentication_service/redirect_issue.html [hunk 3]
```diff
@@ -1,4 +1,4 @@
-{% extends "authentication_service/registration/registration.html" %}
+{% extends "authentication_service/redirect_issue.html" %}
```
Comment (AltusBarry) [inline authentication_service/templates/springster/authentication_service/redirect_issue.html]: See previous comment.

File: authentication_service/templates/springster/authentication_service/registration.html [hunk 4]
```diff
@@ -1,8 +1,8 @@
-{% extends "authentication_service/demo/oops.html" %}
+{% extends "authentication_service/registration.html" %}
```
Comment (AltusBarry) [inline authentication_service/templates/springster/authentication_service/registration.html]: See previous comment.

File: project/settings.py [hunk 5]
```diff
@@ -95,14 +65,60 @@
     "django_otp.middleware.OTPMiddleware",
     "authentication_service.middleware.ThemeManagementMiddleware",
     "authentication_service.middleware.OIDCSessionManagementMiddleware",
-    "authentication_service.middleware.RedirectManagementMiddleware"
+    "authentication_service.middleware.RedirectManagementMiddleware",
+    "crum.CurrentRequestUserMiddleware",
 ]
 
+# TODO: django-layers-hr needs looking into
```
Comment (ruands) [inline project/settings.py]: Already present in `settings_base.py`, should be removed here.

Comment (AltusBarry) [reply project/settings.py]: There is no TODO for django-layers in settings base. Only template loader updates were added in settings_base

Comment (ruands) [reply project/settings.py]: Ah, you are right. My bad.

**Pull request closed without merging**